### ISSUE TITLE
[Bug] "The AWS Access Key Id you provided does not exist in our records"

### ISSUE THREAD

**Issue by jedsundwall** [source-cooperative/source.coop#24]
**Description of Bug:**

I have generated a new API key for Source and added to my AWS Config file at  `~/.aws/config` using the following approach:

```
[profile source]
aws_access_key_id=<REDACTED>
aws_secret_access_key=<REDACTED>
endpoint-url=https://data.source.coop
```

When I try to upload data to my repository using this command: `aws s3 cp og_base.png s3://jed/another-test/ --profile=source`, I receive this error:

> upload failed: ./og_base.png to s3://jed/another-test/og_base.png An error occurred (InvalidAccessKeyId) when calling the PutObject operation: The AWS Access Key Id you provided does not exist in our records.

**Steps to Reproduce:**
- Step 1: Create new Source API key
- Step 2: Try to upload data using AWS s3 cp command

**Expected Behavior:**

I expect the data to be uploaded to my repository.

**Actual Behavior:**

The upload fails

**Screenshots:**

**Additional Context:**

I tested uploading data earlier this week because someone else had reported this error and I was able to upload data without any issues.

**Comment by kbgg** [comment]
@jedsundwall you exposed your admin credentials here, I recommend you revoke the API key and update the issue description to change the credentials to `<REDACTED>`

**Comment by jedsundwall** [comment]
LMAO. I was planning to redact those! Rookie.

**Comment by m-mohr** [comment]
> ```
> [profile source]
> aws_access_key_id=<REDACTED>
> aws_secret_access_key=<REDACTED>
> endpoint-url=https://data.source.coop
> ```
> 
> When I try to upload data to my repository using this command: `aws s3 cp og_base.png s3://jed/another-test/ --profile=source`, I receive this error:


I think you either need to use `[source]` in the config file or "--profile="profile source" in the CLI command.

**Comment by jedsundwall** [comment]
Ugh. I didn't realize the edit history was visible. Keys have been destroyed 💣! 

Also, I figured it out. The issue was that I'd used `endpoint-url` instead of `endpoint_url` – a bit of confusion on my part. I just assumed the flag used in a CLI command should be identical to the parameter in the config file. 

Also FWIW, I'm using a slightly different approach to what we propose on the site.

I've moved my credentials to the `~/.aws/credentials` file with the following:

```
[source]
aws_access_key_id = <REDACTED>
aws_secret_access_key = <REDACTED>
```

I have the rest of the profile set up in ` ~/.aws/config` with the following:

```
[profile source]
endpoint_url = https://data.source.coop
```

I've successfully used `cp`, `sync`, `ls`, and `rm` with this configuration. 

`aws s3 rm s3://jed/another-test/ --profile source`

I've also tested this and it does not work if ` ~/.aws/config` is configured like this:

```
[source]
endpoint_url = https://data.source.coop
```

That is, it as to say `[profile source]`.

Closing this because this was a config error on my part. This is something we need to do some more user testing to see how people tend to configure their CLIs.
